services/housekeeper/housekeeper.go

```diff
@@ -248,6 +273,48 @@ func (hk *Housekeeper) UpdateProposerDutiesWithoutChecks(headSlot uint64) {
 	log.WithField("numDuties", len(_duties)).Infof("proposer duties updated: %s", strings.Join(_duties, ", "))
 }
 
+// When the update is triggered, we store registered validators in Redis for an the current and next epoch.
+// This allows us to maintain 1 epoch in advance, which is enough to check if a validator is registered for a given epoch.
+func (hk *Housekeeper) updateMevCommitValidatorRegistrations(headSlot uint64) {
+	// Only update once per epoch - at the epoch start of the epoch
+	if headSlot%common.SlotsPerEpoch == 0 {
+		return
+	}
+
+	epoch := headSlot / common.SlotsPerEpoch
+
+	duties, err := hk.beaconClient.GetProposerDuties(epoch)
+	if err != nil {
+		hk.log.WithError(err).Error("failed to get proposer duties for epoch")
+		return
+	}
+	nextDuties, err := hk.beaconClient.GetProposerDuties(epoch + 1)
+	if err != nil {
+		hk.log.WithError(err).Error("failed to get proposer duties for next epoch")
+		return
+	}
+	// Combine duties from current and next epoch into a single array
```

Comment:
Don't need this imo and should purely use current epoch (where duties are finalized)

Reply:
I don't think it hurts, in the case of a reorg in proposer duties, we'll have slightly more data in the cache. I think that's worth it for the latency benefits we're getting. Especially for the first validator in the new epoch.